test/bin/s3-server.js

```diff
@@ -73,55 +83,113 @@ function createTempDirForBucket() {
   return tmpDirBucketPath;
 }
 
-/* eslint-disable max-nested-callbacks */
-fs.stat(path, (err, stats) => {
-  if (err || !stats.isDirectory()) {
-    process.exit(1);
+/**
+ * Copies files from the watched folder the the "bucket"
+ * @param {String} bucket
+ */
+function copyFiles(bucket) {
+  const walker = walk.walk(path);
+
+  walker.on('file', (root, fileStats, next) => {
+    const pathToFile = `${Path.relative(path, root)}/${fileStats.name}`;
+    const stream = fs.createReadStream(Path.join(path, pathToFile));
+
+    awsClient.putObject({
+      Bucket: bucket,
+      Key: pathToFile,
+      Body: stream
+    }, (err) => {
+      if (err) {
+        Log.info(err, err.stack);
+      }
+      next();
+    });
+  });
+}
+
+/**
+ * Deletes the contents of the "bucket"
+ * @param {String} bucket
+ */
+function emptyBucket(bucket) {
+  awsClient.listObjects({Bucket: bucket}, (listErr, listData) => {
+    if (listErr) {
+      throw listErr;
+    }
+    listData.Contents.forEach((el) => {
+      awsClient.deleteObject({Bucket: bucket, Key: el.Key}, (deleteErr) => {
+        if (deleteErr) {
+          throw deleteErr;
+        }
+      });
+    });
+  });
+}
+
+/**
+ * Handles cleanup on app exit
+ */
+function exitHandler(err) {
+  let code = 0;
+
+  if (err) {
+    Log.error(err);
+    code = 1;
   }
-  const bucket = getBucketName(path);
-  const tmpDir = createTempDirForBucket();
-
-  const client = new S3rver({
-    port,
-    hostname,
-    silent: false,
-    directory: tmpDir
+  Log.info('Cleaning up temp directory');
+  cleanupTempDir();
+  process.exit(code);
+}
+
+function onFileChange(bucket) {
+  emptyBucket(bucket);
+  copyFiles(bucket);
+}
+
+function createBucket(bucket) {
+  awsClient.createBucket({Bucket: bucket}, (createBucketErr) => {
+    if (createBucketErr) {
+      throw createBucketErr;
+    }
+    copyFiles(bucket);
+
+    // Watch for both change and unlink events.
+    chokidar.watch(path, {persistent: true}).on('change', () => onFileChange(bucket));
+    chokidar.watch(path, {persistent: true}).on('unlink', () => onFileChange(bucket));
   });
+}
 
-  client.run((serverErr, host, p) => {
-    if (serverErr) {
+function init() {
+  fs.stat(path, (err, stats) => {
+    if (err || !stats.isDirectory()) {
+      Log.error('Path is not a directory');
```

Comment:
Do we want to log the path as part of the error message so the user knows which one is invalid?